action_server/frontend/src/components/CancelButton.tsx

```diff
@@ -0,0 +1,44 @@
+import { FC, useMemo } from 'react';
+import { Button, Tooltip } from '@sema4ai/components';
+import { IconStop } from '@sema4ai/icons';
+import { cancelRun } from '~/lib/requestData';
+import { RunStatus } from '~/lib/types';
+
+type Props = {
+  runId?: string;
+  disabled?: boolean;
+  status: RunStatus;
+  requestId?: string;
+};
+
+export const CancelButton: FC<Props> = ({ runId, disabled, status, requestId }) => {
+  // Either runId or requestId must be provided.
+  if (!runId && !requestId) {
```

Comment:
since `runId` and `requestId ` are required, I think they shouldn't be optional in the `Props` type?

Reply:
Well, one of them is required, but if one is set then the other is optional (thus, both need to be marked as optional in the props).